libmscore/splitMeasure.cpp

```diff
@@ -104,8 +104,27 @@ void Score::splitMeasure(Segment* segment)
       Fraction ticks2 = measure->ticks() - ticks1;
       m1->setTimesig(measure->timesig());
       m2->setTimesig(measure->timesig());
-      m1->adjustToLen(ticks1.reduced(), false);
-      m2->adjustToLen(ticks2.reduced(), false);
+      ticks1.reduce();
+      ticks2.reduce();
+      // Now make sure this reduction doesn't go 'beyond' the original measure's
+      // actual denominator for both resultant measures.
```

Comment:
Haven't tested yet, but was thinking about this last night and at least I finally realize why it was the way it was, and in principle this is the right fix.  You can't always preserve the original denominator, like when splitting 4/4 into 3/8 and 5/8, so the original code was needing to build the durations bottom-up.  Then it made sense to reduce, but it shouldn't have reduced all the way.

Presumably a similar thing happens on join, like combining 4/4 and 3/8, the result cannot be expressed in a denominator less than 8.  But, combining 3/8 and 5/8 will be problematic, an algorithm like this will say, keep the 8, when somehow it might be nice to know 4/4 is probably the desired answer.  Except when it isn't.  Hmm, in that case I guess you can still look at the nominal timesig.

Reply:
Hey Marc. Yeah the join functions will be a little bit more complicated in fixing up. I'd hate to leave them alone while fixing this, so I'm willing to take a stab at it, but I also realize it will require a little bit more reasoning behind the decisions here and I'm not sure I'm clear as to what should be done. 

P.S. With this PR, if you were to test Select Similar by Beat filter, it will clearly show the flaw in the old Similar Beat when using the first note of split measures related to regular measures, so these two will go "hand-in-hand" if you will.